### PR TITLE
feat(sdk): `aws.QueueRef`

### PR DESCRIPTION
**Pull request by eladb**
Introduce `aws.QueueRef` as a way to add an external queue reference to an application:

```js
bring aws;
bring cloud;

let myQueue = new aws.QueueRef("arn:aws:sqs:...."); // <-- ARN of an existing SQS queue

new cloud.Function(inflight () => {
  myQueue.push("new message");
});
```

This works both when running in the simulator and when running on AWS.

The Wing Console also includes some metadata about the queue such as the queue URL and the AWS Console URL:

image

### Issues

* Related to #5879 (only implements queue and we need additional resources).
* Related to #5170

### Misc

Added an option `link: bool` to `ui.FIeld` to indicate that the value returned is a link. We should make this a clickable link in the Wing Console.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by MarkMcCulloh** [comment]
Is this a one-off solution for the related issue of referencing existing infra or is this intended to be the pattern to follow going forward?

Side note on the benchmark regressions: I'm willing to bet the additional ~90ms added to all the sim compilations are the imports added to `libs/wingsdk/src/shared-aws/queue.ts`. shared-aws is exported at the root so we'd loading all of these regardless of the target. We've previously avoided this dynamic imports for target-specific stuff, but not sure how easy that'd be here.

**Comment by hasanaburayyan** [comment]
So firstly this is fucking cool, especially what it does in the console 🔥 

Thinking out loud here...

What about organizing these reference classes under a `reference` namespace so something like this:
```js
let outbox = new aws.reference.Queue("arn...");
```

**Comment by eladb** [comment]
@MarkMcCulloh asked:

> Is this a one-off solution for the related issue of referencing existing infra or is this intended to be the pattern to follow going forward?

This is intended to be the pattern going forward for all external references, so we will have `aws.BucketRef`, `aws.TopicRef`, etc...

**Comment by eladb** [comment]
@hasanaburayyan suggested:

> What about organizing these reference classes under a reference namespace so something like this: `aws.reference.Queue`

I am a flat lists and prefixes over namespace hierarchies as it's dramatically improves discoverability.

**Comment by eladb** [comment]
@MarkMcCulloh, what would you recommend to do about the performance regressions (amazing that this was caught)?

File: libs/wingsdk/src/shared-aws/util.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Token } from "cdktf";
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/shared-aws/util.ts:1]: > what would you recommend to do about the performance regressions (amazing that this was caught)?

Without doing any testing, my guess is that the issue is this. queue.ts imports this file, and having this at the top causes a `require("cdktf")` when you do `require("@winglang/sdk")`.

Try moving this inside isValidArn as a `require`

File: examples/tests/sdk_tests/queue/queue-ref.test.w [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    qref.pop();
+  });
+
+  test "can push to an external queue (QueueRef)" {
+    qref.push("m1");
+    qref.push("m2");
+    qref.push("m3");
+    util.waitUntil(() => {
+      return c.peek() == 3;
+    });
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/queue/queue-ref.test.w:30]: Random design thoughts - should `util.waitUntil` throw by default if the condition is never met? (Right now there's a default timeout of around a minute, and if it fails, then it just returns `false`). And perhaps the throwing could be disabled with `throw: false`?

Comment (eladb) [reply examples/tests/sdk_tests/queue/queue-ref.test.w:30]: Mmm? It's supposed to throw after the timeout... Otherwise what's the point?

**Review by hasanaburayyan — APPROVED** [review]
LGTM, Im excited about this direction

File: examples/tests/sdk_tests/queue/queue-ref.test.w (hunk 2 above)
Comment (eladb) [reply examples/tests/sdk_tests/queue/queue-ref.test.w:30]: OK, I am looking at the code and you are right. It's not throwing. Of course it needs to throw!

File: docs/docs/04-standard-library/aws/api-reference.md [hunk 3]
```diff
@@ -146,6 +146,155 @@ The IAM certificate identifier value.
 ---
 
 
+### QueueRef <a name="QueueRef" id="@winglang/sdk.aws.QueueRef"></a>
```
[the remark this answers is not in the document]
Comment (eladb) [reply docs/docs/04-standard-library/aws/api-reference.md:149]: I added a section in `queue.md` with information.

File: examples/tests/sdk_tests/queue/queue-ref.test.w (hunk 2 above)
Comment (eladb) [reply examples/tests/sdk_tests/queue/queue-ref.test.w:30]: https://github.com/winglang/wing/issues/6006

**Pull request merged**